projects/storefrontapp-e2e-cypress/cypress/support/requireLoggedIn.commands.ts

```diff
@@ -0,0 +1,173 @@
+import { generateMail, randomString } from '../helpers/user';
+declare global {
+  namespace Cypress {
+    interface Chainable {
+      /**
+       * Make sure you are logged in. Returns generated email.
+       *
+       * @memberof Cypress.Chainable
+       *
+       * @example
+        ```
+        cy.requireLoggedIn(user)
+        ```
+       */
+      requireLoggedIn: (
+        user?: AccountData,
+        options?: RequireLoggedInOptions
+      ) => Cypress.Chainable<string>;
+    }
+  }
+}
+
+export interface AccountData {
+  user: string;
+  registrationData: RegistrationData;
+}
+
+export interface RegistrationData {
+  firstName: string;
+  lastName: string;
+  password: string;
+  titleCode: string;
+}
+
+export interface RequireLoggedInOptions {
+  alwaysRegister?: boolean;
+}
+
+Cypress.Commands.add(
+  'requireLoggedIn',
+  (accountData?: AccountData, options: RequireLoggedInOptions = {}) => {
+    const apiUrl = Cypress.env('API_URL');
+    const config = {
+      tokenUrl: `${apiUrl}/authorizationserver/oauth/token`,
+      newUserUrl: `${apiUrl}/rest/v2/electronics/users/?lang=en&curr=USD`,
+      client: {
+        client_id: Cypress.env('CLIENT_ID'),
+        client_secret: Cypress.env('CLIENT_SECRET')
+      }
+    };
+
+    function login(uid: string, password: string) {
+      return cy.request({
+        method: 'POST',
+        url: config.tokenUrl,
+        body: {
+          ...config.client,
+          grant_type: 'password',
+          username: uid,
+          password
+        },
+        form: true,
+        failOnStatusCode: false
+      });
+    }
+
+    function loginAsGuest() {
+      return cy.request({
+        method: 'POST',
+        url: config.tokenUrl,
+        body: {
+          ...config.client,
+          grant_type: 'client_credentials'
+        },
+        form: true,
+        failOnStatusCode: false
+      });
+    }
+
+    function registerUser(
+      uid: string,
+      registrationData: RegistrationData,
+      access_token: string
+    ) {
+      return cy.request({
+        method: 'POST',
+        url: config.newUserUrl,
+        body: {
+          firstName: registrationData.firstName,
+          lastName: registrationData.lastName,
+          password: registrationData.password,
+          titleCode: registrationData.titleCode,
+          uid
+        },
+        headers: {
+          Authorization: `bearer ${access_token}`
+        }
+      });
+    }
+
+    function loginAsNewUser(uid: string, password: string) {
```

Comment:
This function doesn't have `failOnStatusCode: false` and it's the only difference to `login()` function. Do we really need 2 functions for login?

Reply:
Added extra param to function and used that one in both places.